src/manager/TickManager.ts

```diff
@@ -1,46 +1,190 @@
-import JSBI from 'jsbi';
+import JSBI from "jsbi";
+import assert from "assert";
 import { Tick } from "../model/Tick";
 import { jsonMapMember, jsonObject } from "typedjson";
 
 @jsonObject
 export class TickManager {
   @jsonMapMember(Number, Tick, { name: "ticks_json" })
+  // key = Tick.tickIndex
   private ticks: Map<number, Tick>;
 
   constructor(ticks: Map<number, Tick> = new Map()) {
     this.ticks = ticks;
   }
 
   get(tickIndex: number): Tick {
-    return this.ticks.get(tickIndex) || new Tick();
+    if (this.ticks.has(tickIndex)) return this.ticks.get(tickIndex)!;
+    const newTick = new Tick(tickIndex);
+    this.set(newTick);
```

Comment:
This may not be proper here because we need `get` method as view function to CorePool. If user call some `get` on uninitialized ticks, then the map will have these tick entries even if they don't have any liquidities. When `getNextInitializedTick` is calling, ticks above will be considerd as initialized ticks and cause mistakes. I will suggest adding an independant get method for view use. For distinction, they can be named as `getAndInitIfAbsent` and `get`.

Reply:
I will add a read-only get method and adjust the names accordingly.